libraries/chain/worker_evaluators.cpp

```diff
@@ -441,4 +439,78 @@ namespace golos { namespace chain {
         });
     }
 
+    void worker_payment_approve_evaluator::do_apply(const worker_payment_approve_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_21__1013, "worker_payment_approve_operation");
+
+        auto approver_witness = _db.get_witness(o.approver);
+        GOLOS_CHECK_LOGIC(approver_witness.schedule == witness_object::top19,
+            logic_exception::approver_of_payment_should_be_in_top19_of_witnesses,
+            "Approver of payment should be in Top 19 of witnesses");
+
+        GOLOS_CHECK_LOGIC(o.state != worker_techspec_approve_state::approve,
+            logic_exception::payment_does_not_need_approving_it_can_be_only_disapproved,
+            "Payment does not need approving, it can be only disapproved");
```

Comment:
Example.
I'm witness.
I see payments for existing work.
I disapprove it.
But some time later I understand that work is really good, and I was mistaken.
I can approve work, and chain should remove my vote from the state.
Why can't I approve work?